core/src/main/codegen/templates/Parser.jj

```diff
@@ -4708,14 +4708,40 @@ SqlNode ArrayConstructor() :
 {
     <ARRAY> { s = span(); }
     (
-        LOOKAHEAD(1)
-        <LPAREN>
-        // by sub query "MULTISET(SELECT * FROM T)"
-        e = OrderedQueryOrExpr(ExprContext.ACCEPT_QUERY)
-        <RPAREN>
```

Comment:
That's a lot of 'validation' code in the parser. That's a code smell. Move it to the validator.

My hunch is that you don't need the `conformance.allowArrayFunction()` at all, anywhere. In the validator you can just look up a function called 'array' in the usual way.

Reply:
Ok, I will remove the conformance. I'm not sure I understand the implementation, do you want to always parse a function call? Then in the `SqlValidator`, when the array function is loaded (via the Spark library) validate that all the arguments are non-query expressions.